src/board_controller/openbci/ganglion_wifi.cpp

```diff
@@ -93,6 +242,80 @@ void GanglionWifi::read_thread ()
 
         double timestamp = get_timestamp ();
         db->add_data (timestamp, package);
-        streamer->stream_data (package, 18, timestamp);
+        streamer->stream_data (package, num_channels, timestamp);
+    }
+    delete[] package;
+}
+
+void GanglionWifi::read_thread_impedance ()
+{
+    int res;
+    unsigned char b[OpenBCIWifiShieldBoard::package_size];
+    double *package = new double[num_channels];
+    for (int i = 0; i < num_channels; i++)
+    {
+        package[i] = 0.0;
+    }
+    while (keep_alive)
+    {
+        // check start byte
+        res = server_socket->recv (b, OpenBCIWifiShieldBoard::package_size);
+        if (res != OpenBCIWifiShieldBoard::package_size)
+        {
+            safe_logger (spdlog::level::warn, "recv result: {}", res);
+            if (res == -1)
+            {
+#ifdef _WIN32
+                safe_logger (spdlog::level::warn, "WSAGetLastError is {}", WSAGetLastError ());
+#else
+                safe_logger (spdlog::level::warn, "errno {} message {}", errno, strerror (errno));
+#endif
+            }
+
+            continue;
+        }
+
+        if (b[0] != START_BYTE)
+        {
+            continue;
+        }
+        int channel_num = (int)(b[1]);
+        // parsing resistance value for channel as asci string
+        int val = 0;
+        int i = 0;
+        // find first non zero byte
+        for (i = 20; i > 2; i--)
+        {
+            if (b[i] != 0)
+            {
+                break;
+            }
+        }
+        std::string asci_value ((const char *)(b + 2), i - 2);
```

Comment:
If i remember, the impedance string was terminated with 'Z'. did you have to account for that somewhere?

Reply:
For ganglion BLE, yes its Z terminated. For Ganglion WIFI just zeroes